app/components/e-select.vue

```diff
@@ -0,0 +1,216 @@
+<template>
+  <div :class="b(modifiers)">
+    <select :value="value"
+            :class="b('select')"
+            :disabled="disabled"
+            v-bind="$attrs"
+            @change="onChange"
+    >
+      <option disabled value="">{{ selectText }}</option>
+      <option v-for="item in optionsList" :key="name + '-' + item.value" :value="item.value">{{ item.label }}</option>
+    </select>
+    <span v-if="!hasDefaultState" :class="b('icon-splitter')"></span>
+  </div>
+</template>
+
+<script>
+  import formStates from '@/mixins/form-states';
+
+  /**
+   * Renders a styled select element. Options can be passed with the `optionsList` property.
+   */
+  export default {
+    name: 'e-select',
+    // components: {},
+    mixins: [formStates],
+    inheritAttrs: false,
+
+    props: {
+      /**
+       * Value for vue model binding.
+       */
+      value: {
+        required: true,
+        type: String,
+      },
+
+      /**
+       * Name is used to generate a unique key in vue.
+       */
+      name: {
+        required: true,
+        type: String,
+      },
+
+      /**
+       * OptionsList defines the options which are rendered in the select.
+       *
+       * e.g. `[{ value: <option-1>, label: <Option 1> },{ value: <option-2>, label: <Option 2>},...]`
+       */
+      optionsList: {
+        required: true,
+        type: Array,
+      },
+
+      /**
+       * The text to display if no option is selected by default.
+       */
+      selectText: {
+        default: 'Please choose...',
+        type: String,
+      }
+    },
+    // data() {
+    //   return {};
+    // },
+
+    computed: {
+      /**
+       * Defines state modifier classes
+       *
+       * @returns {Object} BEM classes
+       */
+      modifiers() {
+        return {
+          ...this.stateModifiers,
+        };
```

Comment:
- [ ] You can include the stateModifiers direct if you don't combine them with other component states